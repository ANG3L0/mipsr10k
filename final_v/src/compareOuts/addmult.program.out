Chronologic VCS simulator copyright 1991-2011
Contains Synopsys proprietary information.
Compiler version E-2011.03_Full64; Runtime version E-2011.03_Full64;  Dec 10 00:51 2012
@@
@@
@@                     0  Asserting System reset......
@@
@@                     0 : System STILL at reset, can't show anything
@@
@@
@@                    10 : System STILL at reset, can't show anything
@@
@@
@@                    20 : System STILL at reset, can't show anything
@@
@@@ Unified Memory contents hex on left, decimal on right: 
@@@
@@@ mem[    0] = 205f0000203f0002 : 2332583132542205954
@@@ mem[    8] = 209f0006207f0004 : 2350597556825686020
@@@ mem[   16] = 23ff000e20bf0008 : 2593791971067625480
@@@ mem[   24] = 4043040740220406 : 4630549271054255110
@@@ mem[   32] = 409f04094c870408 : 4656444977709515784
@@@ mem[   40] = 4c85040840a3040b : 5513817752275715083
@@@ mem[   48] = 4c6404054c220403 : 5504529065352233987
@@@ mem[   56] = 40c7040840650406 : 4667703972279419910
@@@ mem[   64] = 402304024cc80407 : 4621542050536883207
@@@ mem[   72] = 4c42040240670403 : 5494958903062365187
@@@ mem[   80] = 4c42040240420402 : 5494958903059940354
@@@ mem[   88] = 4c42040240420402 : 5494958903059940354
@@@ mem[   96] = 4fff041f40420402 : 5764330580326089730
@@@ mem[  104] = 4022040343ff041f : 4621260579707749407
@@@ mem[  112] = 0000000000000555 : 1365
@@@
@@@
@@                   602 : System halted
@@
@@@ System halted on HALT instruction
@@@
@@
@@  59 cycles / 29 instrs = 2.034483 CPI
@@
@@  1740.00 ns total time to execute
@@

$finish called from file "testbench/pipeline_test.v", line 817.
$finish at simulation time                  602
           V C S   S i m u l a t i o n   R e p o r t 
Time: 602
CPU Time:      0.310 seconds;       Data structure size:   0.3Mb
Mon Dec 10 00:51:03 2012
