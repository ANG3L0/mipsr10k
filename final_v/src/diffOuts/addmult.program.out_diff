< @@@ mem[    0] = 207f1000205f0000 : 2341607923985088512
< @@@ mem[    8] = 4042040840420405 : 4630267800374608901
< @@@ mem[   16] = 4042040440420407 : 4630267783194739719
< @@@ mem[   24] = 40e8040940850406 : 4676992650807936006
< @@@ mem[   32] = b543000040c9040a : 13061283345391485962
< @@@ mem[   40] = 4040340240611403 : 4629757601211552771
< @@@ mem[   48] = f43ffff44041fda1 : 17600067293302357409
< @@@ mem[   56] = 0000000000000555 : 1365
< @@@
< @@@ mem[ 4104] = 0000000000000008 : 8
< @@@ mem[ 4112] = 0000000000000010 : 16
< @@@ mem[ 4120] = 0000000000000018 : 24
< @@@ mem[ 4128] = 0000000000000020 : 32
< @@@ mem[ 4136] = 0000000000000028 : 40
< @@@ mem[ 4144] = 0000000000000030 : 48
< @@@ mem[ 4152] = 0000000000000038 : 56
< @@@ mem[ 4160] = 0000000000000040 : 64
< @@@ mem[ 4168] = 0000000000000048 : 72
< @@@ mem[ 4176] = 0000000000000050 : 80
< @@@ mem[ 4184] = 0000000000000058 : 88
< @@@ mem[ 4192] = 0000000000000060 : 96
< @@@ mem[ 4200] = 0000000000000068 : 104
< @@@ mem[ 4208] = 0000000000000070 : 112
< @@@ mem[ 4216] = 0000000000000078 : 120
> @@@ mem[    0] = 205f0000203f0002 : 2332583132542205954
> @@@ mem[    8] = 209f0006207f0004 : 2350597556825686020
> @@@ mem[   16] = 23ff000e20bf0008 : 2593791971067625480
> @@@ mem[   24] = 4043040740220406 : 4630549271054255110
> @@@ mem[   32] = 409f04094c870408 : 4656444977709515784
> @@@ mem[   40] = 4c85040840a3040b : 5513817752275715083
> @@@ mem[   48] = 4c6404054c220403 : 5504529065352233987
> @@@ mem[   56] = 40c7040840650406 : 4667703972279419910
> @@@ mem[   64] = 402304024cc80407 : 4621542050536883207
> @@@ mem[   72] = 4c42040240670403 : 5494958903062365187
> @@@ mem[   80] = 4c42040240420402 : 5494958903059940354
> @@@ mem[   88] = 4c42040240420402 : 5494958903059940354
> @@@ mem[   96] = 4fff041f40420402 : 5764330580326089730
> @@@ mem[  104] = 4022040343ff041f : 4621260579707749407
> @@@ mem[  112] = 0000000000000555 : 1365
