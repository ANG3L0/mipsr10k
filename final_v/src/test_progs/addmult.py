

if __name__ == '__main__':
   r1=2
   print "r1=2" # lda   $r1, 2
   r2=0
   print "r2=0"# lda   $r2, data1 /* 0 */
   r3=4
   print "r3=4"# lda   $r3, data2 /* 4 */
   r4=6
   print "r4=6"# lda   $r4, data3 /* 6 */
   r5=8
   print "r5=8"# lda   $r5, data4 /* 8 */
   # lda   $r31,data7 /* nothing should happen */
   r6=r1+r2
   print "r6 = " + str(r1+r2)# addq  $r1,$r2,$r6 /* 2+0000 = 0002 -> r6 */
   r7=r2+r3
   print "r7 = " + str(r2+r3)# addq	 $r2,$r3,$r7 /* r7 = 4 */
   r8=r4*r7
   print "r8 = " + str(r4*r7)# mulq  $r4,$r7,$r8 /* r8 = 48 */
   r9=r4
   print "r9 = " + str(r4)# addq  $r4,$r31,$r9 /*r9 = 6 */
   r11=r5+r3
   print "r11 = " + str(r5+r3)# addq  $r5,$r3,$r11
   r8=r4*r5
   print "r8 = " + str(r4*r5)# mulq  $r4,$r5,$r8
   r3 = r2*r1
   print "r3 = " + str(r2*r1)# mulq  $r1,$r2,$r3
   r5 = r3*r4
   print "r5 = " + str(r3*r4)# mulq  $r3,$r4,$r5
   r6 = r3+r5
   print "r6 = " + str(r3+r5)# addq  $r3,$r5,$r6
   r8 = r6+r7
   print "r8 = " + str(r6+r7)# addq  $r6,$r7,$r8
   r7 = r6*r8
   print "r7 = " + str(r6*r8)# mulq  $r6,$r8,$r7
   r2 = r1+r3
   print "r2 = " + str(r1+r3)# addq  $r1,$r3,$r2
   r3 = r7+r3
   print "r3 = " + str(r7+r3)# addq  $r3,$r7,$r3
   r2 = r2*r2
   print "r2 = " + str(r2)# mulq  $r2,$r2,$r2
   r2 = r2+r2
   print "r2 = " + str(r2)# addq  $r2,$r2,$r2
   r2 = r2*r2
   print "r2 = " + str(r2)# mulq  $r2,$r2,$r2
   r2 = r2+r2
   print "r2 = " + str(r2)# addq  $r2,$r2,$r2
   r2 = r2*r2
   print "r2 = " + str(r2)# mulq  $r2,$r2,$r2
   r2 = r2+r2
   print "r2 = " + str(r2)# addq  $r2,$r2,$r2
   print "r31=0"# mulq  $r31,$r31,$r31
   print "r31=0"# addq  $r31,$r31,$r31
   r3=r1+r2
   print "r3 = " + str(r1+r2)# addq  $r1,$r2,$r3
   # call_pal 0x555
