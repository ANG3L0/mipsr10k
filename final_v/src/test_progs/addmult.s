

   data1 = 0x0000
   data2 = 0x0004
   data3 = 0x0006
   data4 = 0x0008
   data5 = 0x000a
   data6 = 0x000c
   data7 = 0x000e
   lda   $r1, 2
   lda   $r2, data1 /* 0 */
   lda   $r3, data2 /* 4 */
   lda   $r4, data3 /* 6 */
   lda   $r5, data4 /* 8 */
   lda   $r31,data7 /* nothing should happen */
   addq  $r1,$r2,$r6 /* 2+0000 = 0002 -> r6 */
   addq	 $r2,$r3,$r7 /* r7 = 4 */
   mulq  $r4,$r7,$r8 /* r8 = 48 */
   addq  $r4,$r31,$r9 /*r9 = 6 */
   addq  $r5,$r3,$r11 /*r11 = 12 */
   mulq  $r4,$r5,$r8
   mulq  $r1,$r2,$r3
   mulq  $r3,$r4,$r5
   addq  $r3,$r5,$r6
   addq  $r6,$r7,$r8
   mulq  $r6,$r8,$r7
   addq  $r1,$r3,$r2
   addq  $r3,$r7,$r3
   mulq  $r2,$r2,$r2
   addq  $r2,$r2,$r2
   mulq  $r2,$r2,$r2
   addq  $r2,$r2,$r2
   mulq  $r2,$r2,$r2
   addq  $r2,$r2,$r2
   mulq  $r31,$r31,$r31
   addq  $r31,$r31,$r31
   addq  $r1,$r2,$r3
   call_pal 0x555
