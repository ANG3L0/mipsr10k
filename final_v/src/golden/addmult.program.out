Chronologic VCS simulator copyright 1991-2012
Contains Synopsys proprietary information.
Compiler version G-2012.09_Full64; Runtime version G-2012.09_Full64;  Dec 10 00:19 2012
@@
@@
@@                     0  Asserting System reset......
@@
@@                     0 : System STILL at reset, can't show anything
@@
@@
@@                   100 : System STILL at reset, can't show anything
@@
@@
@@                   200 : System STILL at reset, can't show anything
@@
@@
@@                   300 : System STILL at reset, can't show anything
@@
@@                   360  Deasserting System reset......
@@
@@
@@@ Unified Memory contents hex on left, decimal on right: 
@@@
@@@ mem[    0] = 205f0000203f0002 : 2332583132542205954
@@@ mem[    8] = 209f0006207f0004 : 2350597556825686020
@@@ mem[   16] = 23ff000e20bf0008 : 2593791971067625480
@@@ mem[   24] = 4043040740220406 : 4630549271054255110
@@@ mem[   32] = 409f04094c870408 : 4656444977709515784
@@@ mem[   40] = 4c85040840a3040b : 5513817752275715083
@@@ mem[   48] = 4c6404054c220403 : 5504529065352233987
@@@ mem[   56] = 40c7040840650406 : 4667703972279419910
@@@ mem[   64] = 402304024cc80407 : 4621542050536883207
@@@ mem[   72] = 4c42040240670403 : 5494958903062365187
@@@ mem[   80] = 4c42040240420402 : 5494958903059940354
@@@ mem[   88] = 4c42040240420402 : 5494958903059940354
@@@ mem[   96] = 4fff041f40420402 : 5764330580326089730
@@@ mem[  104] = 4022040343ff041f : 4621260579707749407
@@@ mem[  112] = 0000000000000555 : 1365
@@@
@@@
@@                 18220 : System halted
@@
@@@ System halted on HALT instruction
@@@
@@
@@  179 cycles / 28 instrs = 6.392857 CPI
@@
@@  5340.00 ns total time to execute
@@

$finish called from file "testbench/testbench.v", line 290.
$finish at simulation time                19220
           V C S   S i m u l a t i o n   R e p o r t 
Time: 1922000 ps
CPU Time:      0.430 seconds;       Data structure size:   0.1Mb
Mon Dec 10 00:19:14 2012
